src/lib/wallets/useEthersSigner.ts

```diff
@@ -1,34 +1,23 @@
-import { JsonRpcSigner, ethers } from "ethers";
-import { useEffect, useState } from "react";
+import { ethers } from "ethers";
+import { UncheckedJsonRpcSigner } from "lib/rpc/UncheckedJsonRpcSigner";
+import { useMemo } from "react";
 import type { Account, Chain, Client, Transport } from "viem";
 import { Config, useConnectorClient } from "wagmi";
 
-async function clientToSigner(client: Client<Transport, Chain, Account>): Promise<JsonRpcSigner> {
+function clientToSigner(client: Client<Transport, Chain, Account>) {
   const { account, chain, transport } = client;
   const network = {
     chainId: chain.id,
     name: chain.name,
     ensAddress: chain.contracts?.ensRegistry?.address,
   };
   const provider = new ethers.BrowserProvider(transport, network);
-  const signer = await provider.getSigner(account.address);
+  const signer = new UncheckedJsonRpcSigner(provider, account.address);
```

Comment:
why is there such change in bigint pr is it related?

Reply:
yes ethers 6 works incorrect with new metamask version and waits for tx completion instead of immediately returning it

more details here https://github.com/ethers-io/ethers.js/issues/4722